### PR TITLE
 [SNAP-931] Use non-secure randomUUID where appropriate

### PR DESCRIPTION
**Pull request by sumwale**
Adding a non-secure version for random UUID adapted from Android UUID.java. Removing the old implementation in DiskBlockManager.

Use the same for file name in DiskBlockManager, Utils methods, WriteAheadLogBackedBlockRDD

### CODE REVIEW

**Comment by sumwale** [comment]
@soubhik-c @kneeraj @suranjan please review

File: core/src/main/scala/org/apache/spark/util/Utils.scala [hunk 1]
```diff
@@ -2230,7 +2232,34 @@ private[spark] object Utils extends Logging {
...
+    temp
+  }
+
+  /**
+   * Returns a path of temporary file which is in the same directory with `path`.
+   */
+  def tempFileWith(parent: String, prefix: String): File = {
+    var temp: File = null
+    var i = 0
+    while (i < 50) {
```
Comment (suranjan) [inline core/src/main/scala/org/apache/spark/util/Utils.scala]: Why 50 times? can't have till we get a file that doesn't exist.

Comment (sumwale) [inline core/src/main/scala/org/apache/spark/util/Utils.scala]: Ok

**Comment by suranjan** [comment]
For my understanding, if this is for performance, I couldn't understand why new one is faster? While loop in previous version?

**Comment by sumwale** [comment]
In Spark version there is no while loop. This is a possible bug in case there is a file name clash but very rare.

As stated in the issue, the new one is faster because it uses java.util.Random rather than the inbuilt one (UUID.randomUUID) which uses java.util.SecureRandom. Latter is slow and can potentially block for indefinite periods since it depends on collection of genuine random data (rather than pseudo-random numbers as used by java.util.Random). In perf analysis it frequently shows up as having a decent share of time. This level of security is not required for file names since the name will retry if present. Please see the issue for more details.

**Comment by suranjan** [comment]
Ok. Looks good.

**Pull request merged**